### PR TITLE
Cluster fix

### PR DESCRIPTION
**Pull request by abdu-n**
Added new variables to variable.tf
Added aurora setting to cluster_config.tf
Replaced analytical_dataset_generator with adg in security_groups.tf
Added hive_metastore_endpoint to configuration.yml.tpl

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -40,3 +40,18 @@ variable "emr_instance_type" {
...
 }
+
+variable "metadata_store_master_username" {
+  description = "Username for metadata store master RDS user"
+  default     = "hive"
+}
+
+variable "metadata_store_pdm_reader_username" {
+  description = "Username for metadata store readonly RDS user"
+  default     = "pdm-reader"
```
Comment (TheDogfather17) [inline variables.tf]: I thought currently we have no requirement for a PDM-READER role, only writer? curretly pdm-reader does not exist.

File: cluster_config/configurations.yaml.tpl [hunk 2]
```diff
@@ -37,10 +37,10 @@ Configurations:
     "hive.compactor.initiator.on": "true"
     "hive.compactor.worker.threads": "1"
     "hive.support.concurrency": "true"
-    "javax.jdo.option.ConnectionURL": "jdbc:mysql://${hive_metastore_fqdn}:3306/hive?createDatabaseIfNotExist=true"
+    "javax.jdo.option.ConnectionURL": "jdbc:mysql://${hive_metastore_endpoint}:3306/hive?createDatabaseIfNotExist=true"
```
Comment (TheDogfather17) [inline cluster_config/configurations.yaml.tpl]: maybe you can do a similar implementation here as seen in adg: https://github.com/dwp/aws-analytical-dataset-generation/blob/7fa2bc80cb9079eda06a01a7b9cf3ff66a9c8fd1/cluster_config/configurations.yaml.tpl#L48

**Review by TheDogfather17 — CHANGES_REQUESTED** [review]

File: cluster_config/configurations.yaml.tpl [hunk 3]
```diff
@@ -37,10 +37,10 @@ Configurations:
     "hive.compactor.initiator.on": "true"
     "hive.compactor.worker.threads": "1"
     "hive.support.concurrency": "true"
-    "javax.jdo.option.ConnectionURL": "jdbc:mysql://${hive_metastore_fqdn}:3306/hive?createDatabaseIfNotExist=true"
+    "javax.jdo.option.ConnectionURL": "jdbc:mysql://${hive_metastore_endpoint}:3306/${hive_metastore_database_name}?createDatabaseIfNotExist=true"
```
Comment (TheDogfather17) [inline cluster_config/configurations.yaml.tpl:40]: you have to pass the value for this to this file otherwise it wont resolve correctly.

**Review by TheDogfather17 — CHANGES_REQUESTED** [review]

File: variables.tf [hunk 4]
```diff
@@ -40,3 +40,13 @@ variable "emr_instance_type" {
     production  = "m5.2xlarge"
   }
 }
+
+variable "metadata_store_master_username" {
```
Comment (UdayChokkam) [inline variables.tf]: is this variable used anywhere else?

**Review by UdayChokkam — APPROVED** [review]

**Review by TheDogfather17 — APPROVED** [review]

**Pull request merged**